File: simple/php/extract-switch-in-method_after.php
function properties($param) {
  $bean = $this->setSetting($param);

  // set other properties...
}

function setSetting($param) {
  $bean = null;
  switch ($param) {
    case "DAO":
      $bean = new DAO();
      break;
    case "DAOStub":
      $bean = new DAOStub();
      break;
    case "AppConfig":
      $bean = new AppConfig();
      break;
    return $bean;
  }
}

File: simple/java/extract-switch-in-method_after.java
void properties(String param) {
  bean = setSetting(param);

  // set other properties...
}

public Bean setSetting(String param) {
  Bean bean;
  switch (param) {
    case "DAO" :
    bean = new DAO();
    break;
  case "DAOStub":
    bean = new DAOStub();
    break;
  case "AppConfig":
    bean = new AppConfig();
    break;
    return bean;
  }
}